MapboxNavigation/RouteManeuverViewController.swift

```diff
@@ -169,13 +169,15 @@ class RouteManeuverViewController: UIViewController {
         }
     }
     
-    func updateStreetNameForStep() {
+    func updateStreetNameForStep(currentLeg: RouteLeg? = nil) {
         if let step = step, step.isNumberedMotorway, let codes = step.codes {
             destinationLabel.unabridgedText = codes.joined(separator: NSLocalizedString("REF_DELIMITER", bundle: .mapboxNavigation, value: " / ", comment: "Delimiter between route numbers in a road concurrency"))
         } else if let name = step?.names?.first {
             destinationLabel.unabridgedText = name
         } else if let destinations = step?.destinations {
             destinationLabel.unabridgedText = destinations.prefix(min(numberOfDestinationLines, destinations.count)).joined(separator: "\n")
+        } else if let currentLeg = currentLeg, let step = step, step.maneuverType == .arrive {
```

Comment:
This case needs to appear above all the other cases. Otherwise, if the step has a `name` (usually just the street name), it’ll override whatever more specific name the developer assigned to the waypoint.